### PR TITLE
[Kenrel] Add FP16 LLaMA YARN rotary_embedding.

### PR DESCRIPTION
**Pull request by changqi1**

### CODE REVIEW

File: src/kernels/rotary_embedding_kernels.h
Comment (pujiang2018) [inline src/kernels/rotary_embedding_kernels.h:1]: looks no modification for this file? can we delete the blank line?

**Comment by pujiang2018** [comment]
@abenmao since it changes LlamaYaRNScaledRotaryEmbedding, could you pls help to review the changes?

File: src/kernels/rotary_embedding_kernels.h
Comment (changqi1) [reply src/kernels/rotary_embedding_kernels.h:1]: 感觉这里需要空行，对齐格式

**Review by abenmao — APPROVED** [review]
LGTM~

**Pull request merged**